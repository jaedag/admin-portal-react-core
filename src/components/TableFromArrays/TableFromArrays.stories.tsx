import React from 'react'
import { Meta, StoryFn } from '@storybook/react'
import TableFromArrays, { TableFromArraysProps } from './TableFromArrays'

export default {
  title: 'Components/TableFromArrays',
  component: TableFromArrays,
  argTypes: {
    loading: { control: 'boolean' },
  },
} as Meta

const heading = [['Field', 'Value', 'Percentage']]
const tableArray = [
  ['First Name', 'John', 0.02],
  ['Last Name', 'Doe', 0.004],
  ['Email', 'johndoe@gmail.com', 30.48],
]

const Template: StoryFn<TableFromArraysProps> = (args) => (
  <TableFromArrays {...args} />
)

export const Default = Template.bind({})
Default.args = {
  tableArray,
  loading: false,
}

export const WithHeading = Template.bind({})
WithHeading.args = {
  heading,
  tableArray,
  loading: false,
}

export const Loading = Template.bind({})
Loading.args = {
  tableArray,
  loading: true,
}

export const WithStyles = Template.bind({})
WithStyles.args = {
  tableArray,
  loading: false,
  variant: 'striped',
  colorScheme: 'teal',
  tdStyle: {
    col: 2,
    style: {
      color: 'red',
      isNumeric: true,
    },
  },
}
