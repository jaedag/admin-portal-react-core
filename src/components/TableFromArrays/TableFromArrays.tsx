import {
  Skeleton,
  Table,
  TableCellProps,
  TableContainer,
  TableProps,
  Tbody,
  Td,
  Th,
  Thead,
  Tr,
} from '@chakra-ui/react'
import React from 'react'

export type TableFromArraysProps = {
  heading?: string[][]
  caption?: string
  tableArray: (string | number)[][]
  loading: boolean
  tdStyle?: {
    col: number
    style: TableCellProps
  }
} & TableProps

const TableFromArrays = ({
  tableArray,
  heading,
  loading,
  tdStyle,
  ...rest
}: TableFromArraysProps) => {
  return (
    <TableContainer>
      <Table {...rest}>
        <Thead>
          {heading?.map((row, rowNum) => (
            <Tr key={rowNum}>
              {row.map((col, colNum) => (
                <Th key={colNum}>{col}</Th>
              ))}
            </Tr>
          ))}
        </Thead>
        <Tbody>
          {tableArray?.map((row, rowNum) => (
            <Tr key={rowNum}>
              {row.map((col, colNum) => {
                let style = {}

                if (colNum === tdStyle?.col) {
                  style = { ...tdStyle?.style }
                }

                return (
                  <Td {...style}>
                    <Skeleton key={colNum} isLoaded={!loading}>
                      {col}
                    </Skeleton>
                  </Td>
                )
              })}
            </Tr>
          ))}
        </Tbody>
      </Table>
    </TableContainer>
  )
}

export default TableFromArrays
