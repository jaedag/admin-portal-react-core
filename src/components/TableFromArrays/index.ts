export { default, TableFromArraysProps } from './TableFromArrays'
