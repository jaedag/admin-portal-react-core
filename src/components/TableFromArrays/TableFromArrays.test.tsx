import React from 'react'
import { render } from '@testing-library/react'
import TableFromArrays from './TableFromArrays'

describe('TableFromArrays', () => {
  const tableArray = [
    ['John', 'Doe', 'john.doe@example.com'],
    ['Jane', 'Smith', 'jane.smith@example.com'],
  ]

  it('matches snapshot when loading is false', () => {
    const { container } = render(
      <TableFromArrays tableArray={tableArray} loading={false} />
    )

    expect(container).toMatchSnapshot()
  })

  it('matches snapshot when loading is true', () => {
    const { container } = render(
      <TableFromArrays tableArray={tableArray} loading={true} />
    )

    expect(container).toMatchSnapshot()
  })
})
