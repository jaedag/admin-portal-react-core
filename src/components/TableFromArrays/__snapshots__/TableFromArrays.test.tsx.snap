// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableFromArrays matches snapshot when loading is false 1`] = `
<div>
  <div
    class="chakra-table__container css-zipzvv"
  >
    <table
      class="chakra-table css-0"
    >
      <thead
        class="css-0"
      />
      <tbody
        class="css-0"
      >
        <tr
          class="css-0"
        >
          <td
            class="css-0"
          >
            <div
              class="chakra-skeleton css-cdkrf0"
            >
              John
            </div>
          </td>
          <td
            class="css-0"
          >
            <div
              class="chakra-skeleton css-cdkrf0"
            >
              Doe
            </div>
          </td>
          <td
            class="css-0"
          >
            <div
              class="chakra-skeleton css-cdkrf0"
            >
              john.doe@example.com
            </div>
          </td>
        </tr>
        <tr
          class="css-0"
        >
          <td
            class="css-0"
          >
            <div
              class="chakra-skeleton css-cdkrf0"
            >
              Jane
            </div>
          </td>
          <td
            class="css-0"
          >
            <div
              class="chakra-skeleton css-cdkrf0"
            >
              Smith
            </div>
          </td>
          <td
            class="css-0"
          >
            <div
              class="chakra-skeleton css-cdkrf0"
            >
              jane.smith@example.com
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`TableFromArrays matches snapshot when loading is true 1`] = `
<div>
  <div
    class="chakra-table__container css-zipzvv"
  >
    <table
      class="chakra-table css-0"
    >
      <thead
        class="css-0"
      />
      <tbody
        class="css-0"
      >
        <tr
          class="css-0"
        >
          <td
            class="css-0"
          >
            <div
              class="chakra-skeleton css-fotqef"
            >
              John
            </div>
          </td>
          <td
            class="css-0"
          >
            <div
              class="chakra-skeleton css-fotqef"
            >
              Doe
            </div>
          </td>
          <td
            class="css-0"
          >
            <div
              class="chakra-skeleton css-fotqef"
            >
              john.doe@example.com
            </div>
          </td>
        </tr>
        <tr
          class="css-0"
        >
          <td
            class="css-0"
          >
            <div
              class="chakra-skeleton css-fotqef"
            >
              Jane
            </div>
          </td>
          <td
            class="css-0"
          >
            <div
              class="chakra-skeleton css-fotqef"
            >
              Smith
            </div>
          </td>
          <td
            class="css-0"
          >
            <div
              class="chakra-skeleton css-fotqef"
            >
              jane.smith@example.com
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
